extensions/crypto/00_webidl.js

```diff
@@ -23,4 +68,164 @@
     "Algorithm",
     algorithmDictionary,
   );
+
+  const RsaKeyGenDictionary = [
+    ...algorithmDictionary,
+    {
+      key: "publicExponent",
+      converter: webidl.converters["BufferSource"],
+      required: true,
+    },
+    {
+      key: "modulusLength",
+      converter: webidl.converters["unsigned long"],
+      required: true,
+    },
+  ];
+
+  webidl.converters["RsaKeyGenParams"] = webidl.createDictionaryConverter(
+    "RsaKeyGenParams",
+    RsaKeyGenDictionary,
+  );
+
+  const RsaHashedKeyGenDictionary = [
+    ...RsaKeyGenDictionary,
+    {
+      key: "hash",
+      converter: webidl.converters["HashAlgorithmIdentifier"],
+      required: true,
+    },
+  ];
+
+  webidl.converters["RsaHashedKeyGenParams"] = webidl.createDictionaryConverter(
+    "RsaHashedKeyGenParams",
+    RsaHashedKeyGenDictionary,
+  );
+
+  const SupportedNamedCurves = ["P-256", "P-384", "P-512"];
+  const EcKeyGenDictionary = [
+    ...algorithmDictionary,
+    {
+      key: "namedCurve",
+      converter: (V, opts) => {
+        const namedCurve = SupportedNamedCurves
+          .find((key) => key.toLowerCase() == V.toLowerCase());
+        if (namedCurve == undefined) {
+          throw new DOMException(
+            "namedCurve not supported",
+            "NotSupportedError",
+          );
+        }
```

Comment:
Don't do this in the webidl validator. Do it after.

Reply:
Done